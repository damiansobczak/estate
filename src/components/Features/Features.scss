@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.features {
  margin: 80px 0;
  border: 1px solid transparent;
  &__desc {
    flex-direction: column;
  }
  &__title {
    color: $color-text-primary;
  }
  &__text {
    color: $color-text-secondary;
    line-height: 1.5;
    font-weight: 500;
    margin-bottom: 50px;
  }
}
