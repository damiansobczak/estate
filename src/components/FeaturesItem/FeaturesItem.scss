@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.featuresItem {
  text-align: center;

  &__icon {
    height: 100px;
    position: relative;
    z-index: 0;
    text-align: center;
    margin-bottom: 20px;
    font-size: 4rem;
    display: flex;
    align-items: center;
    justify-content: center;
    color: $color-accent;

    &:before {
      content: attr(data-number);
      position: absolute;
      top: 50%;
      left: 0;
      right: 0;
      margin: 0 auto;
      transform: translateY(-50%);
      font-size: 142px;
      font-weight: 800;
      color: $color-neutral-200;
      z-index: -1;
    }
  }

  &__title {
    text-align: center;
    font-weight: 700;
    max-width: 160px;
    margin: 0 auto;
    line-height: 1.5;
    color: $color-text-primary;
  }
}
