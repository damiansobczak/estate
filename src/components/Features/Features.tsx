import React from "react";
import "./Features.scss";
import FeaturesItem from "../FeaturesItem/FeaturesItem";

class Features extends React.Component<any, any> {
  render() {
    return (
      <section className="features">
        <div className="container">
          <div className="features__desc columns">
            <h4 className="features__title column is-6-tablet">Cały czas zastanawiasz się dlaczego powinieneś wybrać nasze usługi?</h4>
            <p className="features__text column is-8-tablet">Poznaj kilka faktów o naszej firmie. Mamy nadzieję że rozjaśni to Twoje wątpliwości. Pamiętaj że nasi konsultanci pozostają do Twojej dyspozycji!</p>
          </div>
          <div className="features__items columns">
            <FeaturesItem number="1" icon="key" title="Najniższe ceny w mieście" />
            <FeaturesItem number="2" icon="check-square" title="Wszystko załatwimy za Ciebie" />
            <FeaturesItem number="3" icon="gift" title="Dajemy roczną gwarancję" />
            <FeaturesItem number="4" icon="volume-2" title="Stawiamy na komunikację" />
            <FeaturesItem number="5" icon="unlock" title="Bezpieczeństwo transakcji" />
          </div>
        </div>
      </section>
    );
  }
}

export default Features;
