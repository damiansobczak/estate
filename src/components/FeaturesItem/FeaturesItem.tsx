import React from "react";
import "./FeaturesItem.scss";
import { FeatureItemProps } from "./Interfaces/FeatureItemProps";

class FeaturesItem extends React.Component<FeatureItemProps> {
  render() {
    return (
      <div className="featuresItem column">
        <div className="featuresItem__icon" data-number={this.props.number}>
          <span className={`icon-${this.props.icon}`}></span>
        </div>
        <div className="featuresItem__title">{this.props.title}</div>
      </div>
    );
  }
}

export default FeaturesItem;
