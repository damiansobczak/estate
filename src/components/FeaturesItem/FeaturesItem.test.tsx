import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import FeaturesItem from "./FeaturesItem";
import Features from "../Features/Features";

describe("<FeaturesItem />", () => {
    it("renders component correctly", () => {
        const tree = renderer.create(<FeaturesItem number="1" icon="key" title="Najniższe ceny w mieście" />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it("renders number prop correctly", () => {
        const wrapper = shallow(<FeaturesItem number="1" icon="key" title="Najniższe ceny w mieście" />);
        expect(wrapper.find(".featuresItem__icon").prop("data-number")).toContain("1");
    });

    it("renders icon name correctly", () => {
        const wrapper = shallow(<FeaturesItem number="1" icon="key" title="Najniższe ceny w mieście" />);
        expect(wrapper.find(".featuresItem__icon > span").prop("className")).toContain("icon-key");
    });

    it("renders title correctly", () => {
        const wrapper = shallow(<FeaturesItem number="1" icon="key" title="Najniższe ceny w mieście" />);
        expect(wrapper.find(".featuresItem__title").html()).toContain("Najniższe ceny w mieście");
    });
});
