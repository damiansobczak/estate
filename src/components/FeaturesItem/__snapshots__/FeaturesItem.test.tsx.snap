// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturesItem /> renders component correctly 1`] = `
<div
  className="featuresItem column"
>
  <div
    className="featuresItem__icon"
    data-number="1"
  >
    <span
      className="icon-key"
    />
  </div>
  <div
    className="featuresItem__title"
  >
    Najniższe ceny w mieście
  </div>
</div>
`;
